src/main/scala/edu/berkeley/ce/rockslicing/BlockVTK.scala

```diff
@@ -0,0 +1,213 @@
+      return false
+    }
+
+    // pointA and pointB are on the same line from the center, so check which one is closer to the center
+    val d1 = (pointA._1 - center._1) * (pointA._1 - center._1) + (pointA._2 - center._2) * (pointA._2 - center._2)
+    val d2 = (pointB._1 - center._1) * (pointB._1 - center._1) + (pointB._2 - center._2) * (pointB._2 - center._2)
+    d1 > d2
+  }
+
+  /**
+    * Arranges the vertices of each of the faces of the input block such that they are in a counter-
+    * clockwise orientation relative their face's unit normal
+    * @param block A rock block
+    * @return A mapping from each face of the block to a Seq of vertices for that face arranged in counter-
+    *         clockwise order relative to its unit normal
+    */
+  private def orientVertices(block: Block): Map[Face, Seq[(Double, Double, Double)]] = {
+    val faceVertices = block.findVertices
+    faceVertices.keys.zip(
+      faceVertices.map { case (face, vertices) =>
+        // Rotate vertices to all be in x-y plane
+        val R = BlockVTK.rotationMatrix(face.normalVec, (0.0, 0.0, 1.0))
+        val rotatedVerts = vertices map { vertex =>
+          val rotatedVertex = R * DenseVector[Double](vertex._1, vertex._2, vertex._3)
+          (rotatedVertex(0), rotatedVertex(1), rotatedVertex(2))
+        }
+        // Order vertices in counter-clockwise orientation
+        val center = BlockVTK.findCenter(rotatedVerts)
+        val orderedVerts = {
+          if (face.normalVec._3 < -NumericUtils.EPSILON) {
+            // If z-component of normal vector points in negative z-direction, orientation
+            // needs to be reversed otherwise points will be ordered clockwise
+            rotatedVerts.sortWith(BlockVTK.ccwCompare(_, _, center)).reverse
+          }
+          else {
+            rotatedVerts.sortWith(BlockVTK.ccwCompare(_, _, center))
+          }
+        }
+        // Rotate vertices back to original orientation
+        val invR = R.t // Inverse of rotation matrix is equal to its transpose
+        orderedVerts map { vertex =>
+          val orderedVertex = (invR * DenseVector[Double](vertex._1, vertex._2, vertex._3)).map {
+            NumericUtils.roundToTolerance(_) }
+          (orderedVertex(0), orderedVertex(1), orderedVertex(2))
+        }
+      }
+    ).toMap
+  }
+
+  /**
+    * Finds the indices of the face vertices in the global list of vertices
+    * @param faceOrientedVerts Mapping from each face to a Seq of vertices for that face
+    * @param vertices Seq of distinct vertices for the rock block
+    * @return A mapping from each face of the block to a Seq of integers that represent
+    *         the indices of the vertices in the global vertex list
+    */
+  private def connectivity(faceOrientedVerts: Map[Face, Seq[(Double, Double, Double)]],
+                           vertices: Seq[(Double, Double, Double)]): Map[Face, Seq[Int]] = {
+    faceOrientedVerts.keys.zip(
```

Comment:
Actually, if you want, I think you can use the `map` function directly over a `Map` value to simplify things. So then you can do something like `faceOrientedVerts.map` and avoid the `toMap` at the end.

Check this out: http://stackoverflow.com/questions/9130368/map-both-keys-and-values-of-a-scala-map